### DISCUSSION TITLE
New Callout Component

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#4118] [General] [answered]
Hey team! There's been concern with a few designers on our team about the new, extended Callout component taking up more vertical space than the old version. Is it possible to have the option to move the icon + header on the same line like it was previously?

cc: @char-mainewang & @kevinbtu

**Comment by serifluous** [comment]
Hey @ceyerkes @char-mainewang! Can you share a few examples of layouts where the increased vertical space is a concern?

**Comment by ceyerkes** [comment]
![Screenshot 2024-10-24 at 9 21 58 AM](https://github.com/user-attachments/assets/6d30b05d-dc6b-4cae-a7fc-83af67a2ddcb)

**Reply by ceyerkes** [reply]
wondering if it would be possible to have a condensed version of the callout where everything (icon + header + text) could all be on one line like in Banner? or maybe that's defeating the purpose of the Callout?

**Reply by serifluous** [reply]
Thanks for sharing! Since we've heard this feedback a few times (cc @dreamealist), we're going to update the layout. Here's the ticket in the backlog: https://twilio-productivity.atlassian.net/browse/DS-554

We originally changed the layout to differentiate Callout from Alert even more, and give some breathing room to the close button in the new dismissible variant of Callout. But it sounds like the tradeoff in vertical spacing might not be worth it.

Here's the new proposed layout: https://www.figma.com/design/aP2JnFO5Ds81mTY0iwLxPl/feat(Keyboard-Key)%3A-add-component?node-id=45188-676&t=RIaaitD36JNhEwa6-4 Let us know if this works, or if you have other feedback!

**Reply by ceyerkes** [reply]
I'll ask the team to provide feedback, but i think this is exactly what i was envisioning, thank you so much!

**Reply by char-mainewang** [reply]
just wanted to follow up with another example of where the new call out takes up significant vertical real-estate where it could block other content users may want to see 
![image](https://github.com/user-attachments/assets/5805409d-e0bf-44fe-b12d-d708df48ff20)

**Reply by char-mainewang** [reply]
@serifluous The new design proposals look fantastic! I especially liked the updated left stroke in the new callout component—it’s a clean and simple way to differentiate it from the alert component. I really appreciate your team’s promptness in getting this updated. <3

**Reply by serifluous** [reply]
Awesome, so great to hear! We'll keep y'all updated in this thread and #twilio-ux when the changes go out in code + Figma.

**Comment by robyn-bautista** [comment]
thanks for this response @serifluous, we were going to give the same feedback :) could you give a little more context on the direction of updating the UI to the latest version? on why we went with going with this treatment vs the previous alert treatment?

**Reply by serifluous** [reply]
Yep! You can read the discussion in #3862 that explains some of the history behind Alert and Callout, and why they should be visually distinct from each other.

As part of the explorations to make them more distinct from each other, we decided to keep Alert pretty compact, and Callout a little airier and take up more space with wider padding. But it looks like we went a little _too_ much in that direction, so we're dialing it back a bit now from feedback from teams.

**Comment by serifluous** [comment, marked as answer]
Hey all! This change is now available in code and Figma.

**Comment by ssonajalg** [comment]
The way it hit quite a few of the Twilio Console UI-s => i would definitely argue that this was a bug, and a very visible one at that. I'll open a discussion with the Web platform to see if we could expedite updating the Console Paste version to get the callouts back to normal.

But I hope there's also some takeaways from this for the Paste team. this spent quite a bit of time of quite a few designers and frontend teams - so could we pretty please gather some more feedback and be a bit more conservative when designing such changes?

**Reply by serifluous** [reply]
Hey @ssonajalg , thanks so much for your feedback! We did go through a feedback loop with a small set of Console designers on this, and it unfortunately wasn't raised as an issue in that small group.

That's obviously not ideal, so we'll be much more thoughtful about changes like this in the future, especially when it affects the size and shape of components, and marking them as "design breaking changes" and getting feedback from a broader group. This is definitely a learning for us, and I'm glad you and other folks brought this up.
